### DISCUSSION TITLE
"Font dictionary for /Times-Roman not found" when trying to fill in form

### DISCUSSION THREAD

**Discussion by maltsev** [py-pdf/pypdf#2168] [Q&A]
I'm trying to fill in a form in a PDF. But when I run `writer.update_page_form_field_values()` I get the following error:

```
Font dictionary for /Times-Roman not found.
```

I do have Times New Roman installed.

Here is the code of my script:

```py
from pypdf import PdfReader, PdfWriter
from pprint import pprint

reader = PdfReader("template.pdf", strict=False)
fields = reader.get_form_text_fields()
pprint(fields)  # It successfully prints all the fields

writer = PdfWriter()
writer.append(reader)

writer.update_page_form_field_values(
    writer.pages[0], {"Elevens namn": "John"},
    auto_regenerate=None
)

with open("filled_pypdf.pdf", "wb") as output_stream:
     writer.write(output_stream)
```

My main goal is to have the form filled. It doesn't matter much which font would be used for that. So if it's easier to swap it for another font, that's fine by me.

**Comment by pubpub-zz** [comment]
@maltsev 
can you share your file for analysis

**Reply by maltsev** [reply]
Sure:  [template.pdf](https://github.com/py-pdf/pypdf/files/12552650/template.pdf)

**Comment by pubpub-zz** [comment]
the problem is that without fonts, the program can not establish the transcoding rules. I need to find a default definition that will be used without correct font...🤔
